app/models/representative.rb

```diff
@@ -0,0 +1,20 @@
+class Representative < ActiveRecord::Base
+  has_many :proposals, dependent: :destroy
+  belongs_to :user
+
+  scope :with_name, -> (name) { where("name like ?", "%#{name}%")}
+  scope :with_gender, -> (gender) { where("gender like ?", "%#{gender}%")}
+  scope :candidates, -> { where(type: 'Candidate') }
+  scope :ambassadors, -> { where(type: 'Ambassador') }
+
+  require 'csv'
+
+  def self.search(name)
+    where("name like ?", "%#{name}%")
```

Comment:
A ideia aqui não era usar o escopo `with_name`? Eu, particularmente, acho melhor deixar só o método (e remover o escopo) se ele for utilizado apenas aqui.